openc3/lib/openc3/microservices/reaction_microservice.rb

```diff
@@ -285,29 +281,52 @@ def run_reaction(reaction:)
     end
 
     def run_action(reaction:, action:)
+      reaction_json = reaction.as_json(:allow_nan => true)
+      # Let the frontend know which action is being run
+      # because we can combine commands and scripts with notifications
+      reaction_json['action'] = action['type']
+      notification = {
+        'kind' => 'run',
+        'type' => 'reaction',
+        'data' => JSON.generate(reaction_json),
+      }
+      AutonomicTopic.write_notification(notification, scope: @scope)
+
       case action['type']
+      when 'notify'
+        run_notify(reaction: reaction, action: action)
       when 'command'
         run_command(reaction: reaction, action: action)
       when 'script'
         run_script(reaction: reaction, action: action)
       end
     end
 
+    def run_notify(reaction:, action:)
+      notification = NotificationModel.new(
+        time: Time.now.to_nsec_from_epoch,
+        severity: action['severity'],
+        url: "/tools/autonomic/reactions",
+        title: "#{reaction.name} run",
+        body: action['value']
+      )
+      NotificationsTopic.write_notification(notification.as_json(:allow_nan => true), scope: @scope)
+      @logger.info "ReactionWorker-#{@ident} #{reaction.name} notify action complete, body: #{action['value']}, severity: #{action['severity']}"
```

Comment:
Should this be a @logger.info? It shows up in the LogMessages every time a reaction runs. I also wondered if the ReactionWorker-N should be in the message ... may be a little confusing and does the user care?